### DISCUSSION TITLE
Not able to run owncast with not priveledged systemuser under dabian

### DISCUSSION THREAD

**Discussion by techware01** [owncast/owncast#1043] [Q&A] [answered]
Hey, 
currently I'm having the problem, that streaming with owncast only works when I set the user to root in the systemd service config. 
When I try to set the user and group to a unpriveledged systemuser owncast the following error pops up in transcoder.log:
`
[tcp @ 0x5b59380] Connection to tcp://127.0.0.1:8927 failed: Connection refused
[hls @ 0x5c73b00] Failed to open file 'http://127.0.0.1:8927/0/stream-jtjtKc3Mg1621672251.ts'
av_interleaved_write_frame(): Connection refused
[tcp @ 0x5b5d680] Connection to tcp://127.0.0.1:8927 failed: Connection refused
[hls @ 0x5c73b00] Failed to open file 'http://127.0.0.1:8927/0/stream-jtjtKc3Mg1621672251.ts'
[tcp @ 0x5b5d680] Connection to tcp://127.0.0.1:8927 failed: Connection refused
[tcp @ 0x5b50f40] Connection to tcp://127.0.0.1:8927 failed: Connection refused
[hls @ 0x5c73b00] Failed to open master play list file 'http://127.0.0.1:8927/stream.m3u8'
[hls @ 0x5c73b00] Master playlist creation failed
`
Than the service restarts. 
I don't know whats happening here. Why is owncast trying to access files via tcp on localhost? And why do I need root for this?
I would very much appreciate your help

**Comment by gtxe** [comment, marked as answer]
maybe this helps:

when running owncast as a system service folowing this documentation:
[https://owncast.online/docs/systemservice](https://owncast.online/docs/systemservice/)
it runs with root permissions.

I inserted this lines to the /etc/systemd/system/owncast.service file:
```
User=myuser
Group=myuser
```

But prior to this owncast already created folders and files as root.
It was not possible to stream with OBS to owncast.

I removed this file: /tmp/streampipe.1935
and did chown myuser:myuser /home/myuser/owncast/* -R
restarted the owncast service, then it works again.

**Reply by techware01** [reply]
I did these steps too. But I missed  /tmp/streampipe.flv . After removing it and restaring the service it now also works on my machine. Thank you!

**Reply by ghost** [reply]
I attempted this on my Debian system and it worked perfectly. 
When I attempted it in Ubuntu 20.04lts I got an error when ruinning `systemctl status owncast.service`:

 Process: 4849 ExecStart=/home/username/owncast/owncast (code=exited, status=1/FAILURE)
   Main PID: 4849 (code=exited, status=1/FAILURE)
 
 Where am I supposed to stick these lines ?
>User=myuser
>Group=myuser

In my server (on debian) they are the first lines under the section [Service] 
that one works fine. And it is running as a normal user.

On the Ubuntu server, if I add them the same as above, I get that process exited error.
When I stick them at the end of the file, the service runs as normal, but when I run `systemctl status owncast.service` I get the following:
 `systemd[1]: /etc/systemd/system/owncast.service:14: Unknown key name 'User' in section 'Install', ignoring.`
` systemd[1]: /etc/systemd/system/owncast.service:15: Unknown key name 'Group' in section 'Install', ignoring.`

When I say as normal, I mean that it is still running as root, which is not what I wanted. 

 Can anyone help me on this? Thanks for your time.

**Reply by ghost** [reply]
For anyone who is also running into this issue. This is how I fixed it. Might be redundant but whatever..

What you need to do is first stop the owncast service with this commmand: 
`sudo systemctl stop owncast.service`
Then `nano /etc/systemd/system/owncast.service`  and put the User and Group under the [Service] section
Then you need to run  `chown -R user:user ~/owncast `
You also need to make sure to do the same as above for the directory `owncast/data/hls` because it refuses to change ownership to the user despite the -R flag 
then run `sudo systemctl daemon-reload`
and then `sudo systemctl enable owncast && sudo systemctl start owncast`
`sudo systemctl restart owncast.service` for good measure

I didnt have to run /tmp/streampipe.1935

**Comment by gabek** [comment]
Glad you got it working!  There's nothing about Owncast that requires root privileges, so sounds like you just had a permissions issue.
